core/src/main/java/io/questdb/griffin/engine/ops/AlterOperation.java

```diff
@@ -266,44 +270,45 @@ private void applyAttachPartition(TableWriter tableWriter) throws SqlException {
         for (int i = 0, n = longList.size(); i < n; i++) {
             long partitionTimestamp = longList.getQuick(i);
             try {
-                int statusCode = tableWriter.attachPartition(partitionTimestamp);
-                switch (statusCode) {
-                    case StatusCode.OK:
-                        break;
-                    case StatusCode.CANNOT_ATTACH_MISSING_PARTITION:
-                        throw putPartitionName(
-                                SqlException.$(tableNamePosition, "attach partition failed, folder '"),
-                                tableWriter.getPartitionBy(),
-                                partitionTimestamp)
-                                .put("' does not exist");
-                    case StatusCode.TABLE_HAS_SYMBOLS:
-                        throw SqlException.$(tableNamePosition, "attaching partitions to tables with symbol columns not supported");
-                    case StatusCode.PARTITION_EMPTY:
-                        throw putPartitionName(
-                                SqlException.$(tableNamePosition, "failed to attach partition '"),
-                                tableWriter.getPartitionBy(),
-                                partitionTimestamp)
-                                .put("', data does not correspond to the partition folder or partition is empty");
-                    case StatusCode.PARTITION_ALREADY_ATTACHED:
-                        throw putPartitionName(
-                                SqlException.$(tableNamePosition, "failed to attach partition '"),
-                                tableWriter.getPartitionBy(),
-                                partitionTimestamp)
-                                .put("', partition already attached to the table");
-                    default:
-                        throw putPartitionName(
-                                SqlException.$(tableNamePosition, "attach partition  '"),
-                                tableWriter.getPartitionBy(),
-                                partitionTimestamp)
-                                .put(statusCode);
+                StatusCode statusCode = tableWriter.attachPartition(partitionTimestamp);
+                if (statusCode != StatusCode.OK) {
+                    throw putPartitionName(
```

Comment:
simpler code, if status is ok happy days, otherwise I use the value of the enum to inform the user, rather than translating it, which will be more difficult to maintain.

Reply:
Are you going to document the status codes? Codes are more maintanance.

Reply:
Are status codes not readable enough?. I mean, if you get `errno=9` you always need to look it up, but these "status codes" are self explanatory?. I could embellish the log line using instead int status codes and a switch case, but I felt this approach was more scalable. I will need to indeed document attach/detach and possible error scenarios.